Autopilot/Path Manager/PathManager.c

```diff
@@ -667,6 +674,19 @@ void copyGPSData(){
     pmData.checkbyteDMA2 = generatePMDataDMAChecksum2();
 }
 
+static void checkForFirstGPSLock(){
+	static char gpsLockFlag = 1;
+	
+	if(((getErrorCodes() & BOR_RESET) || (getErrorCodes() & POR_RESET)) && gpsLockFlag){
```

Comment:
Also the way you placed your if statement, getErrorCodes() will always be called twice, no mater whether gpsLockFlag is 1 or 0. The arrangement of variables in an if do matter. So in your case:
```c```
if(((getErrorCodes() & BOR_RESET) || (getErrorCodes() & POR_RESET)) && gpsLockFlag)
```
The first statement will always be executed (the 2 getErrorCodes() calls), even if gpsLockFlag is 0.

However in this case:
```c
if(gpsLockFlag && ((getErrorCodes() & BOR_RESET) || (getErrorCodes() & POR_RESET)))
```
If gpsLockFlag is 0, neither of the functions will be called.

http://stackoverflow.com/questions/1331318/how-an-if-a-b-statement-is-evaluated